.github/workflows/dev-builds.yml

```diff
@@ -0,0 +1,37 @@
+name: Development builds
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_publish:
+    name: Build source distribution
+    if: github.repository == 'aesara-devs/aesara'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+      - name: Install dependencies and customize pyproject.toml
+        run: |
+          # Download dasel to modify pyproject.toml
+          curl -sSLf https://github.com/TomWright/dasel/releases/download/v2.0.2/dasel_linux_amd64 \
+            -L -o /tmp/dasel && chmod +x /tmp/dasel
+
+          # Modify pyproject.toml to set the nightly version in the form of
+          # x.y.z.postN, where N is the number of commits since the last release
+          /tmp/dasel put -f pyproject.toml project.name -v aesara-nightly
+          /tmp/dasel put -f pyproject.toml tool.setuptools_scm.version_scheme -v post-release
+          /tmp/dasel put -f pyproject.toml tool.setuptools_scm.local_scheme -v no-local-version
```

Comment:
This looks neat!

Reply:
Great, then let's keep this in favor of the Python script-based approach. (Will squash the commit which introduces this.)

Reply:
Squashed. What do you think about the dev builds being called `aesara-nightly` while being built on merge-to-main?

Reply:
That sounds good.